addons/hr_expense/models/hr_expense.py

```diff
@@ -552,7 +583,8 @@ def refuse_sheet(self, reason):
     def approve_expense_sheets(self):
         if not self.user_has_groups('hr_expense.group_hr_expense_user'):
             raise UserError(_("Only HR Officers can approve expenses"))
-        self.write({'state': 'approve', 'responsible_id': self.env.user.id})
+        responsible = self.user_id.id or self.env.user.id
```

Comment:
Actually you should override responsible if not already set. What is the intended behavior ?